src/deploy.ts

```diff
@@ -445,6 +456,31 @@ export async function deploy(
   Telemetry.record({event: "deploy", step: "finish"});
 }
 
+async function findOldestSourceAge(effects: DeployEffects, config: Config): Promise<number> {
+  let oldestAge = -Infinity;
+  const nowMs = Date.now();
+
+  try {
+    for await (const file of effects.visitFiles(config.root, {ignoreObservable: false})) {
+      let stat: Stats;
+      const joinedPath = join(config.root, file);
+      try {
+        stat = await effects.stat(joinedPath);
+        if (stat?.isFile()) {
```

Comment:
This shouldn’t be necessary:

- `effects.stat` always returns a `Promise<Stat>`, so the `?.` shouldn’t be needed
- `effects.visitFiles` only visits files, so the `stat.isFile` shouldn’t be needed

I suppose technically `visitFiles` can return symlinks (and other weird things like block devices maybe?) but that isn’t really intended. As you might guess from the name `visitFiles`, I only intended to enumerate files.